.github/workflows/rust.yml

```diff
@@ -0,0 +1,94 @@
+name: CI
+
+on:
+  push:
+    branches:
+      - '**'
+  pull_request:
+    branches:
+      - '**'
```

Comment:
I don't think the `**` filter is needed... Could it be like hyper, testing pull requests and master pushes?

Reply:
I can change it, but i do not see the draw back of having builds on any kind of push or pull request.
If you make a pull request towards a version branch that is too to be merged to master, you get no ci checks before merge.